### PR TITLE
[Glimmer] Port component elementid test.

### PR DESCRIPTION
**Pull request by jheth**
Moved over the one elementid test and added another one for custom elementid.

Issue #13127

### CODE REVIEW

File: packages/ember-glimmer/tests/integration/components/curly-components-test.js [hunk 1]
```diff
@@ -586,4 +586,30 @@ moduleFor('Components test: curly components', class extends RenderingTest {
...
+    let FooBarComponent = Component.extend({
+      tagName: 'h1'
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: 'something' });
+
+    this.render('{{foo-bar id="custom-id"}}');
+
+    var foundId = this.$('h1').attr('id');
+    assert.equal(foundId, 'custom-id');
```
Comment (jheth) [inline packages/ember-glimmer/tests/integration/components/curly-components-test.js]: This is failing for glimmer2. It still generates an emberXX id.

**Comment by rwjblue** [comment]
Looks great! 

@chancancode - r?

File: packages/ember-htmlbars/tests/integration/component_element_id_test.js [hunk 2]
```diff
@@ -1,46 +0,0 @@
...
-  },
-
-  teardown() {
-    runDestroy(owner);
-    runDestroy(view);
-    owner = view = null;
-  }
-});
-
-QUnit.test('passing undefined elementId results in a default elementId', function() {
```
Comment (mixonic) [inline packages/ember-htmlbars/tests/integration/component_element_id_test.js]: Ported to `@test passing undefined elementId results in a default elementId`

**Comment by rwjblue** [comment]
@jheth - this looks great, I think merging #13236 caused some rebase issues here (and added at least one `elementId` test).  Can you double check and rebase?

**Comment by jheth** [comment]
@rwjblue  I've rebased and pushed.  I moved my tests up in the file to be next to the other elementid test.

**Pull request merged**